### PR TITLE
Skip ShadowSQLiteConnection test on M1

### PR DESCRIPTION
**Pull request by utzcoz**
Fix https://github.com/robolectric/robolectric/issues/6967.

### CODE REVIEW

**Comment by utzcoz** [comment]
@hoisie I have tested it on my Linux laptop and M1 laptop, and it is ready for initial reviewing, although I think there are things we can improve.

**Comment by utzcoz** [comment]
@hoisie could we improve the priority of this PR? Looks like one potential contributor of GSoC is using Apple M1 pro.

**Review by hoisie — APPROVED** [review]

File: shadows/framework/src/main/java/org/robolectric/shadows/util/SQLiteLibraryLoader.java [hunk 1]
```diff
@@ -77,6 +81,13 @@ public boolean isLoaded() {
     return loaded;
   }
 
+  public static boolean isOsSupported() {
```
Comment (utzcoz) [inline shadows/framework/src/main/java/org/robolectric/shadows/util/SQLiteLibraryLoader.java:84]: @hoisie could you accept adding this public API?

Comment (utzcoz) [reply shadows/framework/src/main/java/org/robolectric/shadows/util/SQLiteLibraryLoader.java:83]: I will merge it firstly, and we can improve it at later commits before next release.

**Pull request merged**